If the `world-info` has no comment, show it's keys in the placeholder.

DeclineThyself opened the pull request:
If the `world-info` has no comment, show it's keys in the placeholder.
This currently updates only on load.
If you approve of this feature, I'll make the placeholder update on key entry.

<img width="547" height="416" alt="world-info" src="https://github.com/user-attachments/assets/e0cfb9ac-e2fc-481b-ab99-d962beb4507f" />

## Checklist:

- [x] I have read the [Contribution guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

DeclineThyself commented:
> ❗ This PR is against the `release` branch.
> Please make sure this was intended, and you did not want to target the `staging` branch. Only hotfixes, readme changes and similar should be made against `release`.
> You can change the target branch **without recreating the PR** by clicking "Edit" at the top of the page.

Fixed.

DeclineThyself commented:
/gemini review

DeclineThyself commented:
/gemini summary

DeclineThyself commented on public/scripts/world-info.js:
```diff
@@ -3206,6 +3206,11 @@ export async function getWorldEntry(name, data, entry) {
+
+    //If there's no comment, show keys in the placeholder.
+    const keys = data.entries[entry.uid]['key'].join(', ').slice(0,50);
+    keys !== '' && commentInput.attr('placeholder', keys);
```
The comment may be deleted. This is fine as is.

Cohee1207 commented:
I think it's fine as an idea. However, there's a button to backfill comments from keys, but it only takes the first key from the array. You might wanna update it to

DeclineThyself commented:
> I think it's fine as an idea. However, there's a button to backfill comments from keys, but it only takes the first key from the array. You might wanna update it to

How do you want it updated?
Backfill with all keys?

Commit:
Show keys in commentInput's placeholder.

DeclineThyself commented on public/scripts/world-info.js:
> I think it's fine as an idea. However, there's a button to backfill comments from keys, but it only takes the first key from the array. You might wanna update it to

How do you want it updated? Backfill with all keys?


```diff
diff --git a/public/scripts/world-info.js b/public/scripts/world-info.js
index 6109c4496..6da479fc2 100644
--- a/public/scripts/world-info.js
+++ b/public/scripts/world-info.js
@@ -2423,7 +2423,7 @@ async function displayWorldEntries(name, data, navigation = navigation_option.no
         let counter = 0;
         for (const entry of Object.values(data.entries)) {
             if (!entry.comment && Array.isArray(entry.key) && entry.key.length > 0) {
-                entry.comment = entry.key[0];
+                entry.comment = entry.key.join(', ').slice(0, 50);
                 setWIOriginalDataValue(data, entry.uid, 'comment', entry.comment);
                 counter++;
             }

```

DeclineThyself replied:
I forgot to click submit on the review two days ago.

Cohee1207 commented on public/scripts/world-info.js:
```diff
@@ -3189,6 +3192,17 @@ function handleEntryKillSwitchHelper({ entryKillSwitch, entry, data, name, templ
+function setCommentPlaceholder(keys, commentInput) {
+    //Show keys in the placeholder.
+    keys = keys.slice(0,1000);
+    commentInput.attr('placeholder', (keys !== '' ? keys : 'Entry Title/Memo'));
```
The added text is not translatable.

Also it will be easier to do it like

```suggestion
    commentInput.attr('placeholder', (keys || t`Entry Title/Memo`));
```

DeclineThyself replied:
Thank you.
https://github.com/SillyTavern/SillyTavern/pull/4681/commits/e9528faef8ea356b41619c16baae0f34577bec07

Cohee1207 commented on public/scripts/world-info.js:
```diff
@@ -3189,6 +3192,17 @@ function handleEntryKillSwitchHelper({ entryKillSwitch, entry, data, name, templ
+ */
+function setCommentPlaceholder(keys, commentInput) {
+    //Show keys in the placeholder.
+    keys = keys.slice(0,1000);
```
What is the significance of this? Troll prevention? How was the number chosen? 1000 is much more than could fit into even the widest screen.

DeclineThyself replied:
1000 is much more than could fit into even the widest screen, and less then what may cause rendering slowdowns. 
If someone decides to use an entire dictionary as keys to match a language, truncation is safer.

Cohee1207 commented on public/scripts/world-info.js:
```diff
@@ -2881,6 +2881,9 @@ function enableKeysInputHelper({ template, entry, entryPropName, originalDataVal
                 $(this).toggleClass('empty', !data.entries[uid][entryPropName].length);
             }
+            //Update the commentInput's placeholder.
+            const commentInput = $(_event.currentTarget).closest('.world_entry_form').find('textarea[name="comment"]');
```
This update will not run if isFancyInput is true. ie. if the keys is using a select2-styled control

DeclineThyself replied:
Fixed.
https://github.com/SillyTavern/SillyTavern/pull/4681/commits/e9528faef8ea356b41619c16baae0f34577bec07

Cohee1207 commented on public/scripts/world-info.js:
```diff
@@ -3206,6 +3220,11 @@ export async function getWorldEntry(name, data, entry) {
     const commentInput = headerTemplate.find('textarea[name="comment"]');
+
+    //Update the commentInput's placeholder.
+    const keys = data.entries[entry.uid]['key'].join(', ');
```
```suggestion
    const keys = entry.key.join(', ');
```

DeclineThyself replied:
Thank you.
https://github.com/SillyTavern/SillyTavern/pull/4681/commits/e9528faef8ea356b41619c16baae0f34577bec07

Commit:
Fixed.
https://github.com/SillyTavern/SillyTavern/pull/4681#discussion_r2450822592
https://github.com/SillyTavern/SillyTavern/pull/4681#discussion_r2450827302
https://github.com/SillyTavern/SillyTavern/pull/4681#discussion_r2450834282
https://github.com/SillyTavern/SillyTavern/pull/4681#discussion_r2450844011

DeclineThyself commented:
/gemini review

Cohee1207 committed:
Merge branch 'staging' into feat/world-info-comment-placeholder

Cohee1207 committed:
Update comment backfill logic

Cohee1207 committed:
Don't update placeholder with secondary keys

DeclineThyself commented on public/scripts/world-info.js:
```diff
@@ -2423,7 +2424,7 @@ async function displayWorldEntries(name, data, navigation = navigation_option.no
         for (const entry of Object.values(data.entries)) {
             if (!entry.comment && Array.isArray(entry.key) && entry.key.length > 0) {
-                entry.comment = entry.key[0];
+                entry.comment = entry.key.join(', ').slice(0, MAX_COMMENT_LENGTH);
```
Placeholder text does not increase the textbox's size, while the comment value does.
A higher max length here may cause each world info entry to expand due to word wrapping.

Cohee1207 replied:
However, backfill value being different from the placeholder may look weird

Cohee1207 replied:
I can propose cutting down the max length to 100. That should be enough for most users

DeclineThyself replied:
I [initially](https://github.com/SillyTavern/SillyTavern/pull/4681/files/06c05a2e1d426cee7814a4806354b523051f6da4#r2449883185) suggested 50 to show one line before truncation, 100 would show two lines maximum.

I have no preference, I do not use the feature.

Cohee1207 replied:
100 it is, then. I mean, there won't be that much of a reason to use it now.

Cohee1207 committed:
Fix MAX_COMMENT_LENGTH to enforce a limit of 100 characters
